experimental/AlgebraicStatistics/src/DiscreteGraphicalModels.jl

```diff
@@ -0,0 +1,326 @@
+@attributes mutable struct DiscreteGraphicalModel{T, L} <: GraphicalModel{T, L}
+  graph::T
+  labelings::L
+  states::Vector{Int}
+  varnames::Dict{Symbol,VarName}
+  function DiscreteGraphicalModel(G::Graph{S}, states::Vector{Int}, varnames::Dict{Symbol, VarName}) where S <: GraphTypes
+    graph_maps = NamedTuple(_graph_maps(G))
+    graph_maps = isempty(graph_maps) ? nothing : graph_maps
+    return new{Graph{S}, typeof(graph_maps)}(G, graph_maps, states, varnames)
+  end
+end
+
+###################################################################################
+#
+#       Undirected Discrete Graphical Models
+#
+###################################################################################
+
+@doc raw"""
+    discrete_graphical_model(G::Graph{Undirected}, states::Vector{Int}; t_var_name::String="t")
+
+The parametric statistical model associated to an undirected graph.
+It contains an undirected graph `G`, a MarkovRing `S` where the vanishing ideal of the model naturally lives, 
+and a parameter ring whose variables `t[C](i_1, i_2, ..., i_#C)` correspond to potential functions of each clique. 
+
+``` jldoctest 
+julia> M = discrete_graphical_model(graph_from_edges([[1,2], [2,3]]), [2,2,2])
+Discrete Graphical Model on a Undirected graph with 3 nodes and 2 edges with states [2, 2, 2]
+```
+"""
+function discrete_graphical_model(G::Graph{Undirected}, states::Vector{Int}; t_var_name::String="t", p_var_name::String="p")
+  @req length(states) == n_vertices(G) "need one and only one random variable in MarkovRing for each graph vertex"
+  DiscreteGraphicalModel(G, states, Dict{Symbol, VarName}(:t => t_var_name, :p => p_var_name))
+end
+
+varnames(M::DiscreteGraphicalModel) = M.varnames
+states(M::DiscreteGraphicalModel) = M.states
+n_states(M::DiscreteGraphicalModel) = length(states(M))
+maximal_cliques(M::DiscreteGraphicalModel) = maximal_cliques(graph(M))
+
+function Base.show(io::IO, M::DiscreteGraphicalModel{T, L}) where {T, L}
+  io = pretty(io)
+  if is_terse(io)
+    print(io, "Discrete Graphical Model on a")
+    !(L == Nothing) && print(io, " labelled")
+    print(io, " $T")
+  else
+    print(io, "Discrete Graphical Model on a $(graph(M))")
+  end
+  print(io, " with states ", string(M.states))
+end
+
+@doc raw"""
+    @attr Tuple{ModelRing, GenDict} model_ring(M::DiscreteGraphicalModel; cached=false)
```

Comment:
```suggestion
    model_ring(M::DiscreteGraphicalModel; cached=false)
```
or
```suggestion
    model_ring(M::DiscreteGraphicalModel; cached=false) -> Tuple{ModelRing, GenDict}
```


The same applies to other similar uses elsewhere.